tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -154,6 +157,105 @@ public static void fastStartSpan(String operation) {
         fastStartSpan(operation, SpanType.LOCAL);
     }
 
+    /**
+     * Like {@link #startSpan(String, SpanType)}, but does not set or modify tracing thread state.
+     * This is an internal utility that should not be called directly outside of {@link DetachedSpan}.
+     */
+    static DetachedSpan detachInternal(String operation, SpanType type) {
+        Trace maybeCurrentTrace = currentTrace.get();
+        String traceId = maybeCurrentTrace != null
+                ? maybeCurrentTrace.getTraceId() : Tracers.randomId();
+        boolean sampled = maybeCurrentTrace != null
+                ? maybeCurrentTrace.isObservable() : sampler.sample();
+        return new DefaultDetachedSpan(operation, type, traceId, getParentSpanId(maybeCurrentTrace), sampled);
+    }
+
+    private static Optional<String> getParentSpanId(@Nullable Trace trace) {
+        if (trace != null) {
+            Optional<OpenSpan> maybeOpenSpan = trace.top();
+            if (maybeOpenSpan.isPresent()) {
+                return Optional.of(maybeOpenSpan.get().getSpanId());
+            }
+        }
+        return Optional.empty();
+    }
+
+    private static final class DefaultDetachedSpan implements DetachedSpan {
+
+        // Complete the current span.
+        private static final SpanToken DEFAULT_TOKEN = Tracer::fastCompleteSpan;
+
+        private final AtomicBoolean completed = new AtomicBoolean();
+        private final boolean sampled;
+        private final String traceId;
+        private final OpenSpan openSpan;
+
+        DefaultDetachedSpan(
+                String operation, SpanType type, String traceId, Optional<String> parentSpanId, boolean sampled) {
+            this.traceId = traceId;
+            this.sampled = sampled;
+            this.openSpan = OpenSpan.builder()
+                    .parentSpanId(parentSpanId)
+                    .spanId(Tracers.randomId())
+                    .operation(operation)
+                    .type(type)
+                    .build();
+        }
+
+        @Override
+        public SpanToken attach(String operationName, SpanType type) {
+            warnIfCompleted("startSpanOnCurrentThread");
+            Trace maybeCurrentTrace = currentTrace.get();
+            Trace trace = Trace.of(sampled, traceId);
+            setTrace(trace);
+            Tracer.fastStartSpan(operationName, type);
```

Comment:
You also will want to explicitly pass the spanId of `openSpan` as the parentId of the new span